modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,408 @@
+		$extra_class    = ' has-transparency ';
+	}
+
+	if ( isset( $image_meta['dominant_color'] ) ) {
+		if ( ! isset( $attr['style'] ) ) {
+			$attr['style'] = '--dominant-color: #' . $image_meta['dominant_color'] . ';';
+		} else {
+			$attr['style'] .= '--dominant-color: #' . $image_meta['dominant_color'] . ';';
+		}
+		$attr['data-dominant-color'] = $image_meta['dominant_color'];
+
+		$extra_class .= ( wp_color_is_light( $image_meta['dominant_color'] ) ) ? 'dominant-color-light' : 'dominant-color-dark';
+
+		if ( isset( $attr['class'] ) && ! array_intersect( explode( ' ', $attr['class'] ), explode( ' ', $extra_class ) ) ) {
+			$attr['class'] = $extra_class . ' ' . $attr['class'];
+		} else {
+			$attr['class'] = $extra_class;
+		}
+	}
+
+	return $attr;
+}
+add_filter( 'wp_get_attachment_image_attributes', 'wp_tag_add_adjust_to_image_attributes', 10, 2 );
+
+/**
+ * Filter image tags in content to add the dominant color to the image.
+ *
+ * @param string $filtered_image The filtered image.
+ * @param string $context        The context of the image.
+ * @param int    $attachment_id  The attachment ID.
+ *
+ * @return string image tag
+ */
+function wp_tag_add_adjust( $filtered_image, $context, $attachment_id ) {
+
+	$image_meta = wp_get_attachment_metadata( $attachment_id );
+
+	/**
+	 * Controls if the dominant color code should update image meta if not set.
+	 * When an image shown on the site and the meta is not set, the meta will be updated.
+	 *
+	 * @param bool $add_dominant_color_to_image true to add the dominant color to the image: default true.
+	 * @param int  $attachment_id
+	 */
+	if ( ! isset( $image_meta['dominant_color'] ) && apply_filters( 'enable_dominant_color_back_fill', true, $attachment_id ) ) {
+
+		/**
+		 * Controls if the dominant color code should update image meta if not set.
+		 * When an image shown on the site and the meta is not set, the meta will be updated.
+		 *
+		 * @param bool $add_dominant_color_on_pageload
+		 * set to true save dominant color on the fly as part of the page load.
+		 * Set to false schedule the missing color to be set via cron.
+		 * @param int  $attachment_id
+		 */
+		if ( apply_filters( 'enable_dominant_color_back_fill_realtime', false, $attachment_id ) ) {
+			$image_meta = wp_back_fill_dominant_color( $attachment_id, $image_meta );
+		} else {
+			wp_schedule_single_event( time(), 'wp_back_fill_dominant_color', array( $attachment_id ) );
+		}
+	}
+
+	/**
+	 * Allow developers to filter the control the adding of dominant color to the image.
+	 * set to false inorder disable adding the dominant color to the image.
+	 *
+	 * @param bool $add_dominant_color_to_image
+	 * @param int  $attachment_id
+	 * @param array $image_meta
+	 * @param string $filtered_image
+	 * @param string $context
+	 */
+	if ( apply_filters( 'enable_dominant_color_for_image', isset( $image_meta['dominant_color'] ), $attachment_id, $image_meta, $filtered_image, $context ) ) {
```

Comment:
Related to the above: If `dominant_color` or `has_transparency` isn't set in the attachment's metadata, I think it'd be safer to bail early and not even run this filter, at least for the MVP.

In that case, we could set the default here simply to a hard-coded `true`.